src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1089,19 +1036,72 @@ public void SetEditorCellVisibility(bool shown)
     /// <summary>
     ///   Changes the number of mutation points left. Should only be called by MicrobeEditorAction
     /// </summary>
-    internal void ChangeMutationPoints(int change)
+    internal void UpdateMutationPoints()
     {
         if (FreeBuilding || CheatManager.InfiniteMP)
             return;
 
-        MutationPoints = (MutationPoints + change).Clamp(0, Constants.BASE_MUTATION_POINTS);
+        MutationPoints = MutationPointsAfterChange(CurrentSpecies);
 
         gui.UpdateMutationPointsBar();
     }
 
+    internal int MutationPointsAfterChange(MicrobeSpecies newSpecies)
+    {
+        if (FreeBuilding)
+        {
+            return Constants.BASE_MUTATION_POINTS;
+        }
+
+        var rigidityChangeCost = (int)Math.Round(
+            Math.Abs(newSpecies.MembraneRigidity - startingSpecies.MembraneRigidity)
+            * Constants.MEMBRANE_RIGIDITY_COST_PER_STEP * Constants.MEMBRANE_RIGIDITY_SLIDER_TO_VALUE_RATIO);
+
+        var membraneChangeCost = startingSpecies.MembraneType.Equals(newSpecies.MembraneType) ?
+            0 :
+            newSpecies.MembraneType.EditorCost;
+
+        // Calculate which organelles are moved, added, or deleted
+        var commonOrganelles = startingSpecies.Organelles.Organelles
+            .Where(x => newSpecies.Organelles.Organelles.Contains(x));
```

Comment:
I'd use `o` as the letter in every LINQ statement here because you loop over organelles.